subsys/nfc/lib/platform.c

```diff
@@ -64,46 +65,55 @@ nrfx_err_t nfc_platform_setup(void)
 }
 
 
-static nrfx_err_t nfc_platform_tagheader_get(uint8_t index,
-					     uint32_t *tag_header)
+static nrfx_err_t nfc_platform_tagheaders_get(uint32_t tag_header[3])
 {
-	const uint32_t tag_header_addresses[] = {
-#ifdef CONFIG_TRUSTED_EXECUTION_NONSECURE
-		(uint32_t) &NRF_FICR_S->NFC.TAGHEADER0,
-		(uint32_t) &NRF_FICR_S->NFC.TAGHEADER1,
-		(uint32_t) &NRF_FICR_S->NFC.TAGHEADER2
+	__IOM FICR_NFC_Type *ficr_nfc =
+#if defined(NRF_TRUSTZONE_NONSECURE)
```

Comment:
I'm not familiar with NRF_TRUSTZONE_NONSECURE.

Why use it instead of CONFIG_TRUSTED_EXECUTION_NONSECURE?

Reply:
That is the MDK define, meaning that this code is executing in non-secure context.
It is always set when `CONFIG_ARM_NONSECURE_FIRMWARE` is enabled.

`zephyr_compile_definitions_ifdef(CONFIG_ARM_NONSECURE_FIRMWARE NRF_TRUSTZONE_NONSECURE)`.

I'm not sure if there is any difference now between `CONFIG_TRUSTED_EXECUTION_NONSECURE` and `CONFIG_ARM_NONSECURE_FIRMWARE`.

If I remember correctly I think I did this for a reason, but right now I can't recall exactly why. I'll have to look into it a bit more.

Reply:
No reason to believe there is anything wrong with NRF_TRUSTZONE_NONSECURE